backend/sirius-components-compatibility/src/main/java/org/eclipse/sirius/components/compatibility/services/diagrams/DiagramDescriptionNodeAndEdgeDescriptionsPopulator.java

```diff
@@ -162,7 +164,9 @@ private boolean isEnabledByDefault(Layer layer) {
                     childVariableManager.put(NEW_CONTAINER, newContainer.get());
                     childVariableManager.put(NEW_VIEW_CONTAINER, newViewContainer.get());
                     childVariableManager.put(ToolProvider.ELEMENT, self);
-
+                    if (optionalEnvironment.isPresent()) {
+                        childVariableManager.put(Environment.ENVIRONMENT, optionalEnvironment.get());
+                    }
```

Comment:
The changes in this file are not necessary, a child variable manager can access the variable of its parent variable manager. We don't propagate `self` everywhere for example